corehq/apps/api/resources/v0_5.py

```diff
@@ -1044,3 +1057,149 @@ def prepend_urls(self):
+    }
+
+    @staticmethod
+    def to_obj(action_times):
+        '''
+        Takes a flat dict and returns an object
+        '''
+        return namedtuple('action_times', list(action_times))(**action_times)
+
+    def alter_list_data_to_serialize(self, request, data):
+        data['meta']['local_date_timezone'] = self.local_timezone.zone
+        params = request.GET.copy()  # Makes params mutable for creating next_url below
+
+        if data['meta']['total_count'] >= data['meta']['limit']:
+            last_object = data['objects'][-1]
+            params['cursor_local_date'] = last_object.data['local_date']
+            params['cursor_user'] = last_object.data['user']
+
+            next_url = '{}?{}'.format(request.path, params.urlencode())
+            data['meta']['next'] = next_url
+        return data
+
+    def dehydrate(self, bundle):
+        bundle.data['user_id'] = get_user_id_by_username(bundle.data['user'])
+        return bundle
+
+    def obj_get_list(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        params = self._process_params(domain, bundle.request.GET)
+
+        results = self.cursor_query(domain, self.local_timezone, params)
+        return list(map(self.to_obj, results))
+
+    def _process_params(self, domain, params):
+        processed_params = {}
+
+        for param in params:
+            val = params.get(param)
+            if param == 'users':
+                val = params.getlist(param)
+            processed_params[param] = val
+
+            if param == 'limit':
+                processed_params['limit'] = self._process_limit(val)
+
+        if 'local_timezone' in processed_params:
+            self.local_timezone = pytz.timezone(processed_params['local_timezone'])
+        else:
+            self.local_timezone = Domain.get_by_name(domain).get_default_timezone()
+
+        return processed_params
+
+    def _process_limit(self, limit):
+        try:
+            limit = int(limit) or self._meta.limit
+            if limit < 0:
+                raise ValueError
+        except (ValueError, TypeError):
+            raise BadRequest(_('limit must be a positive integer.'))
+
+        if limit > self._meta.max_limit:
+            raise BadRequest(_('Limit may not exceed {}.').format(self._meta.max_limit))
+        return limit
+
+    @classmethod
+    def cursor_query(cls, domain: str, local_timezone: pytz.tzinfo.DstTzInfo,
+                    params: dict = {}) -> list:
+        if 'limit' not in params:
+            params['limit'] = cls._meta.limit
+        queryset = cls._query(domain, local_timezone, params)
+
+        cursor_local_date = params.get('cursor_local_date')
+        cursor_user = params.get('cursor_user')
+
+        if cursor_local_date and cursor_user:
+            queryset = queryset.filter(
+                Q(local_date__gt=cursor_local_date)
+                | (Q(local_date=cursor_local_date) & Q(user__gt=cursor_user))
```

Comment:
Are the parenthesis her in the correct place? I expected the date filters to be grouped separately from the user filter.

Reply:
Yes I believe they are. The results will be ordered by local_date, then by user. For example in the (date,user) ordered pairs below with the cursor pointing to (1,1, albert). 

`(Q(local_date=cursor_local_date) & Q(user__gt=cursor_user))` gives you (1/1,bob)
`Q(local_date__gt=cursor_local_date)` gives you everything after 1/1

(1/1, albert), <-- cursor
(1/1, bob),
(1/2, albert),
(1/2, bob)

Reply:
ah, got it. Thanks